File: 2025/05 May/01 Pascal Triangle/main.cpp
// Link: https://www.geeksforgeeks.org/problems/pascal-triangle0652/1

#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> nthRowOfPascalTriangle(int n)
    {
        n -= 1;
        vector<int> res;
        int prev = 1;
        res.push_back(prev);

        for (int i = 1; i <= n; ++i)
        {
            int curr = (prev * (n - i + 1)) / i;
            res.push_back(curr);
            prev = curr;
        }

        return res;
    }
};

int main()
{
    Solution obj;
    int n = 4;
    vector<int> result = obj.nthRowOfPascalTriangle(n);
    for (int val : result)
    {
        cout << val << " ";
    }
    cout << endl;

    n = 5;
    result = obj.nthRowOfPascalTriangle(n);
    for (int val : result)
    {
        cout << val << " ";
    }
    cout << endl;

    n = 1;
    result = obj.nthRowOfPascalTriangle(n);
    for (int val : result)
    {
        cout << val << " ";
    }
    cout << endl;

    return 0;
}

File: 2025/05 May/01 Pascal Triangle/main.py
# Link: https://www.geeksforgeeks.org/problems/pascal-triangle0652/1


class Solution:
    def nthRowOfPascalTriangle(self, n):
        n -= 1
        res = []
        prev = 1
        res.append(prev)

        for i in range(1, n + 1):
            curr = (prev * (n - i + 1)) // i
            res.append(curr)
            prev = curr

        return res


if __name__ == "__main__":
    obj = Solution()
    n = 4
    print(obj.nthRowOfPascalTriangle(n))
    n = 5
    print(obj.nthRowOfPascalTriangle(n))
    n = 1
    print(obj.nthRowOfPascalTriangle(n))
